lighthouse-core/test/report/html/renderer/snippet-list-renderer-test.js

```diff
@@ -0,0 +1,163 @@
+      lineNumber: 1,
+      message: 'Error',
+    }]);
+
+    // 5 lines are visible, 1 is collapsed
+    assert.equal(collapsedContentLines.length, 1);
+    // All available lines are shown on expansion
+    assert.equal(contentLines.length, 6);
+    // 100 lines in total, so lines towards the end won't be shown
+    const lastLine = contentLines[contentLines.length - 1];
+    assert.equal(lastLine.nextSibling.textContent.trim(), '…');
+
+    // Shows highlight message for second line
+    assert.equal(highlightLines[0].textContent.trim(), 'Error');
+    assert.equal(highlightLines[0].previousSibling.textContent.replace(/\s/g, ''), '1L1');
+  });
+
+  it('Renders first few lines if there are no highlights', () => {
+    const {
+      uncollapsedContentLines,
+      omittedLinesIndicatorsWhenExpanded,
+      omittedLinesIndicatorsWhenCollapsed,
+    } = renderSnippet([]);
+    const lastUncollapsedLine = uncollapsedContentLines[uncollapsedContentLines.length - 1];
+
+    // Shows first 5 visible lines
+    assert.equal(lastUncollapsedLine.textContent.replace(/\s/g, ''), '5L5');
+    // "..." after the available lines, but only shows in expanded state
+    assert.equal(omittedLinesIndicatorsWhenExpanded.length, 1);
+    assert.equal(omittedLinesIndicatorsWhenCollapsed.length, 0);
+  });
+
+  it('Renders first few lines if a highlight is not line-specific', () => {
+    const {uncollapsedContentLines} = renderSnippet([{
+      lineNumber: null,
+      message: 'General error',
+    }]);
+    const lastUncollapsedLine = uncollapsedContentLines[uncollapsedContentLines.length - 1];
+
+    // Shows first 5 visible lines
+    assert.equal(lastUncollapsedLine.textContent.replace(/\s/g, ''), '5L5');
+  });
+
+  it('Renders snippet with multiple highlights surrounded by other lines', () => {
+    const {
+      collapsedContentLines,
+      omittedLinesIndicatorsWhenCollapsed,
+      omittedLinesIndicatorsWhenExpanded,
+    } = renderSnippet([{
+      lineNumber: 40,
+      message: 'Error 1',
+    }, {
+      lineNumber: 70,
+      message: 'Error 2',
+    }], [
+      {
+        from: 30,
+        to: 50,
+      },
+      {
+        from: 60,
+        to: 80,
+      },
+    ]);
+
+    // first available line is collapsed
+    assert.equal(collapsedContentLines[0].textContent.replace(/\s/g, ''), '30L30');
+
+    // puts omitted lines placeholder between the two highlights
+    assert.equal(omittedLinesIndicatorsWhenCollapsed.length, 1);
+    // puts omitted lines placeholder between the two highlights and around the whole snippet
+    assert.equal(omittedLinesIndicatorsWhenExpanded.length, 3);
+  });
+
+  it('Can render both line-specific and non line-specific messages in one snippet', () => {
+    const {highlightLines} = renderSnippet([{
+      message: 'General error',
+    }, {
+      lineNumber: 5,
+      message: 'Error on line',
+    }]);
+
+    assert.equal(highlightLines.length, 2);
+  });
+});
```

Comment:
seems like a few more tests are need. The list as a whole (I think), header, toggling collapsed state (I think you can `.click()` in jsdom), `truncated`...any other states?

Can wait on some of these until the rest of the PR is settled so you don't have to keep changing them